x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java

```diff
@@ -54,6 +66,22 @@ protected AcknowledgedResponse read(StreamInput in) throws IOException {
     @Override
     protected void masterOperation(Task task, DeleteEnrichPolicyAction.Request request, ClusterState state,
                                    ActionListener<AcknowledgedResponse> listener) throws Exception {
+        List<PipelineConfiguration> pipelines = IngestService.getPipelines(state);
+        for (PipelineConfiguration pc: pipelines) {
+            Pipeline pipeline = ingestService.getPipeline(pc.getId());
+
+            for (Processor processor: pipeline.getProcessors()) {
+                if (processor instanceof AbstractEnrichProcessor) {
+                    AbstractEnrichProcessor enrichProcessor = (AbstractEnrichProcessor) processor;
+                    if (enrichProcessor.getPolicyName().equals(request.getName())) {
```

Comment:
nit: can we treat the policy names case-insensitive (e.g. equalsIgnoreCase)

Reply:
we do not do this anywhere else best of my knowledge, so PUT /policyA and PUT /policya will create 2... its probably a good idea to change this tho

Reply:
https://github.com/elastic/elasticsearch/pull/44438/files#diff-a37fe0a3791b5e0c8215af76056efd9eR76

The equals ignore case is here.